src/common/time/src/error.rs

```diff
@@ -40,14 +40,25 @@ pub enum Error {
 
     #[snafu(display("Timestamp arithmetic overflow, msg: {}", msg))]
     ArithmeticOverflow { msg: String, location: Location },
+
+    #[snafu(display("Invalid time zone offset: {hours}:{minutes}"))]
+    InvalidTimeZoneOffset { hours: i32, minutes: u32 },
+
+    #[snafu(display("Invalid offset string {raw}: {source}"))]
+    ParseOffsetStr { raw: String, source: ParseIntError },
```

Comment:
Better add a `Location` field since `ParseOffsetStr` occurs multiple times in code.